### PR TITLE
Initialize EB grid data on restart.

### PR DESCRIPTION
**Pull request by atmyers**
Fixing (another) restart bug.

### CODE REVIEW

File: Source/Diagnostics/WarpXIO.cpp [hunk 1]
```diff
@@ -334,6 +334,8 @@ WarpX::InitFromCheckpoint ()
         }
     }
 
+    InitializeEBGridData(maxLevel());
+
```
Comment (PhilMiller) [inline Source/Diagnostics/WarpXIO.cpp:338]: Does the EB only exist on the finest level? If not, should this be in the loop over levels just above?

Comment (atmyers) [reply Source/Diagnostics/WarpXIO.cpp:338]: It only exists on the finest level right now - in `InitializeEBGridData`, the first thing it does is return if `lev != maxLevel()`.

**Comment by ax3l** [comment]
@PhilMiller @peterscherpelz @KZhu-ME or @lgiacome : does one of you like to add a restart test that uses EB? :)

**Comment by peterscherpelz** [comment]
@atmyers You get your first pull request: https://github.com/atmyers/WarpX/pull/1 which adds the restart test in response to @ax3l 's note.

I did confirm it failed on development branch; Bx did not match between original and restart.

Sorry for the delay - my first time working with WarpX tests and I managed to hit a bunch of stupid problems in a row.

**Review by peterscherpelz — APPROVED** [review]

**Review by ax3l — APPROVED** [review]
Thank you both, superb :rocket: :sparkles:

**Pull request merged**